src/bot/functions/bot-utility.service.ts

```diff
@@ -20,4 +20,24 @@ export class BotUtilityService{
       }
     }
   }
+
+  async sendMessageToChannel(channelId: string, payload: APIEmbed | string) {
+    const channel = await this.client.channels.fetch(channelId);
+    if (channel.isTextBased()) {
```

Comment:
pense à faire des sortie au plus tot pour ne pas te retrouver avec des if chainé ^^

Exemple ici :

if(!channel.isTextBased()) {
    this.logger.error(`Channel ${channelId} is not text based`);
    return;
}

const textChannel = channel as TextChannel;
# suite du code